### DISCUSSION TITLE
Is there a way to horizontally scale milvus without k8s? Is there a simple guide?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#18131] [Q&A and General discussion]
Is there a way to horizontally scale milvus without k8s? Is there a simple guide?
For instance if there is not enough RAM on one system.

**Comment by xiaofan-luan** [comment]
emm, you can actually use the script to start new querynode,datanode,indexnode yourself. But the trick is if node crashed you have to handle it your self.
Ansible is yet another deploy way, check the document here https://milvus.io/docs/v2.0.x/install_cluster-ansible.md

**Reply by martinenkoEduard** [reply]
Thanks!
Btw by default does milvus store all data in RAM and then flushes it on HDD?

Will different nodes have different data and then combine input for FLAT index?

**Reply by martinenkoEduard** [reply]
Can you please explain in layman term how does it distribute data between nodes if I use Flat index (for 100% return rate)

**Reply by xiaofan-luan** [reply]
We split data into segment. 
Each segment is flushed into Minio.
Querynode load and balance data by segment.
When search happened, we fetch topK from each querynode and reduce again in proxy.
See https://milvus.io/docs/v2.0.x/architecture_overview.md for more details

**Reply by martinenkoEduard** [reply]
Thanks for the simple explanation!
Now I understand the workflow!
